src/HealthChecks.UI/ServiceCollectionExtensions.cs

```diff
@@ -69,11 +69,29 @@ public static IServiceCollection AddHealthChecksUI(this IServiceCollection servi
 
             if (kubernetesDiscoverySettings.Enabled)
             {
+                KubernetesClientConfiguration kubernetesConfig;

```

Comment:
@ggmaresca Can we hide all this logic in an extension method called AddKubernetesServiceDiscovery?
So we can clear all this configuration setup from here and use it above of:

```csharp
 AddSingleton<IKubernetes>(new Kubernetes(kubernetesConfig))
```

Reply:
```csharp
 if (kubernetesDiscoverySettings.Enabled) { 
  services.AddKubernetesDiscoveryService(kubernetesDiscoverySettings);
}
```